agilent_infiniium_scope.py

```diff
@@ -31,360 +31,389 @@
-        Channel must be one of 0 (AUX), 1, 2, 3, 4.
+    @setting(131, channel='i', level='v', returns='v{level}')
+    def trigger_at(self, c, channel, level=None):
+        """Get or set the trigger source and voltage for edge mode triggering.
+
+        Channel must be one of 0 (AUX), 1, 2, 3, 4, or 5 (LINE).
+        If trigger source is 5 (LINE) the level will be ignored
         """
         dev = self.selectedDevice(c)
-        if channel==0:
+        if channel not in [0, 1, 2, 3, 4, 5]:
+            raise ValueError('Invalid trigger channel: {}'
+                             'Valid channels are [0, 1, 2, 3, 4, 5]'
+                             '0 for AUX, 5 for LINE'.format(channel))
+        if channel == 0:
             channel = 'AUX'
+        elif channel == 5:
+            channel = 'LINE'
         elif isinstance(channel, int):
-            #channel = 'CHAN%d' %channel
-            pass
-        if channel not in [0,1,2,3,4]:
-            raise Exception('Select valid trigger channel')
-        if level is None:
-            resp = yield dev.query('TRIG:LEV? CHAN%s' %channel)
+            channel = 'CHAN{}'.format(channel)
+        yield dev.write('TRIG:EDGE:SOUR {}'.format(channel))
+        if channel != 'LINE':
+            # set trigger level
+            yield dev.write('TRIG:LEV {}, {}'.format(channel, level))
+            resp = yield dev.query('TRIG:LEV? {}'.format(channel))
         else:
-            yield dev.write('TRIG:EDGE:SOUR CHAN%s' %channel) #set channel up for edge triggering  
-            yield dev.write('TRIG:LEV CHAN%s,%f' %(channel,level)) #set trigger level
-            resp = yield dev.query('TRIG:LEV? CHAN%s' %channel)
+            resp = 0.0
         level = float(resp)
         returnValue(level)
 
-    @setting(132, slope = 's', returns = ['s'])
-    def trigger_mode(self, c, slope = None):
+    @setting(132, slope='s', returns=['s'])
+    def trigger_mode(self, c, slope=None):
         """Change trigger mode. Use 'EDGE' for edge triggering.
-        Must be one of 'COMM','DEL','EDGE','GLIT','PATT','PWID','RUNT','SEQ',',SHOL','STAT','TIM','TRAN','TV',',WIND','SBUS1','SBUS2','SBUS3','SBUS4'.
+
+        Must be one of 'COMM', 'DEL', 'EDGE', 'GLIT', 'PATT', 'PWID', 'RUNT',
+        'SEQ', ',SHOL', 'STAT', 'TIM', 'TRAN', 'TV', ',WIND', 'SBUS1', 'SBUS2',
+        'SBUS3', 'SBUS4'.
         """
         dev = self.selectedDevice(c)
         if slope is None:
             resp = yield dev.query('TRIG:MODE?')
         else:
             slope = slope.upper()
-            if slope not in ['COMM','DEL','EDGE','GLIT','PATT','PWID','RUNT','SEQ',',SHOL','STAT','TIM','TRAN','TV',',WIND','SBUS1','SBUS2','SBUS3','SBUS4']:
+            if slope not in ['COMM', 'DEL', 'EDGE', 'GLIT', 'PATT', 'PWID',
+                             'RUNT', 'SEQ', ',SHOL', 'STAT', 'TIM', 'TRAN',
+                             'TV', ',WIND', 'SBUS1', 'SBUS2', 'SBUS3', 'SBUS4']:
```

Comment:
I had noticed that for ',SHOL' too, but was lazy and didn't check the manual to make sure it was actually a type ;).  But now I did, and it is a typo.

Fixed now.